### PR TITLE
Solved the first questions of homework 3

### PR DESCRIPTION
**Pull request by luisfilipels**

### CODE REVIEW

File: src/Assignment3/sorting/SortEx1.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+}
+
+fun main() {
+    val testArray = intArrayOf(5, 10, 5, 20, 5, 5, 10)
+
+    val sort = SortEx1()
+    sort.threePartitionSort(testArray)
+    println(testArray.contentToString())
```
Comment (kunwang555) [inline src/Assignment3/sorting/SortEx1.kt]: var expectedValue = intArrayOf(5, 5, 5, 5, 10, 10, 20)
var actualValue = testArray
assure.equals(expectedValue, actualValue)

Comment (kunwang555) [reply src/Assignment3/sorting/SortEx1.kt]: https://kotlinlang.org/docs/jvm-test-using-junit.html#create-a-test

**Review by kunwang555 — CHANGES_REQUESTED** [review]
Please add some test cases

File: src/Assignment3/sorting/SortEx1.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class SortEx1 {
+
+    fun swap(array: IntArray, left: Int, right: Int) {
+        val swap = array[left]
+        array[left] = array[right]
+        array[right] = swap
+    }
+
+    fun threePartitionSort(array: IntArray) {
```
Comment (kunwang555) [inline src/Assignment3/sorting/SortEx1.kt:13]: add comments on the method/function to tell what the purpose of the method. or link to the question.

**Comment by luisfilipels** [comment]
Added test cases and comments to the first question. Moving on to the next questions.

**Comment by kunwang555** [comment]
@luisfilipels Please let me know once you finished all the changes, and I will review it again. overall those test cases looks good to me

**Comment by luisfilipels** [comment]
Added tests and comments to all sorting-related questions. Is it ok to merge?

**Review by kunwang555 — APPROVED** [review]
please merge.

**Pull request merged**